oci/layout/oci_transport.go

```diff
@@ -72,6 +73,27 @@ func ParseReference(reference string) (types.ImageReference, error) {
 	return NewReference(dir, image)
 }
 
+// NewReferenceWithIndex returns an OCI reference for a directory and a image， sourceIndex points to an image.
+//
+// We do not expose an API supplying the resolvedDir; we could, but recomputing it
+// is generally cheap enough that we prefer being confident about the properties of resolvedDir.
+func NewReferenceWithIndex(dir, image string, sourceIndex int) (types.ImageReference, error) {
```

Comment:
NewReference() Should be calling this function, rather then duplicating the code.